Adyen/Service/TerminalApiAsyncService.cs

```diff
@@ -0,0 +1,88 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Adyen.ApiSerialization;
+using Adyen.Model.TerminalApi;
+using Adyen.Model.TerminalApi.Message;
+using Adyen.Security;
+using Adyen.Service.Resource.Terminal;
+
+namespace Adyen.Service
+{
+    /// <summary>
+    /// Service that sends requests to the Adyen Cloud Terminal API `/async` endpoint.
+    /// </summary>
+    public interface ITerminalApiAsyncService
+    {
+        /// <summary>
+        /// Sends an encrypted <see cref="SaleToPOIRequest"/> or <see cref="SaleToPOIMessage"/> to the terminal-api `/async` endpoint.
+        /// </summary>
+        /// <param name="saleToPoiRequest"><see cref="SaleToPOIRequest"/> or <see cref="SaleToPOIMessage"/>.</param>
+        /// <param name="encryptionCredentialDetails"><see cref="EncryptionCredentialDetails"/>. These must match the credentials that you configured in the Customer Area. Make sure your terminal is updated to latest version.</param>
+        /// <param name="cancellationToken"><see cref="CancellationToken"/>.</param>
+        /// <returns>A <see cref="Task{TValue}"/> that represents the <see cref="string"/>.</returns>
+        Task<string> RequestEncryptedAsync(SaleToPOIMessage saleToPoiRequest, EncryptionCredentialDetails encryptionCredentialDetails, CancellationToken cancellationToken = default);
+        
+        /// <summary>
+        /// Sends a <see cref="SaleToPOIRequest"/> or <see cref="SaleToPOIMessage"/> to the terminal-api `/async` endpoint.
+        /// </summary>
+        /// <param name="saleToPoiRequest"><see cref="SaleToPOIRequest"/> or <see cref="SaleToPOIMessage"/>.</param>
+        /// <param name="cancellationToken"><see cref="CancellationToken"/>.</param>
+        /// <returns>A <see cref="Task{TValue}"/> that represents the <see cref="string"/>.</returns>
+        Task<string> RequestAsync(SaleToPOIMessage saleToPoiRequest, CancellationToken cancellationToken = default);
+        
+        
+        /// <summary>
+        /// Used to decrypt the notification received.
+        /// </summary>
+        /// <param name="notification">Notification.</param>
+        /// <param name="encryptionCredentialDetails"><see cref="EncryptionCredentialDetails"/>.</param>
+        /// <returns>Decrypted notification.</returns>
+        string DecryptNotification(string notification, EncryptionCredentialDetails encryptionCredentialDetails);
+    }
+
+    public class TerminalApiAsyncService : AbstractService, ITerminalApiAsyncService
```

Comment:
Why not specify this is the cloud service in the name? ie
```suggestion
    public class TerminalApiCloudAsyncService : AbstractService, ITerminalApiCloudAsyncService
```

Reply:
I ensured that the URL endpoints match the service name for consistency in the library. I noticed that the library keeps the same naming-conventions for i.e. `Payments`-Api, or `Modifications`. In this case, I prefer to use the `terminal-api-` + `/sync` & `/async` suffixes respectively.


<img width="398" alt="image" src="https://github.com/user-attachments/assets/c920a640-f21f-4d7f-8640-34af9315cfae">

Reply:
Right right, well since the local doesn't have async/sync (correct me if I'm wrong) that should be clear enough